test/cppsuite/tests/csuite_style_example_test.cxx

```diff
@@ -0,0 +1,169 @@
+
+    /* Set the tracing level for the logger component. */
+    logger::trace_level = LOG_INFO;
+
+    /* Printing some messages. */
+    logger::log_msg(LOG_INFO, "Starting " + progname);
+    logger::log_msg(LOG_ERROR, "This could be an error.");
+
+    /* Create a connection, set the cache size and specify the home directory. */
+    const std::string conn_config = std::string(CONNECTION_CREATE) + ",cache_size=500MB";
+    const std::string home_dir = std::string(DEFAULT_DIR) + '_' + progname;
+    connection_manager::instance().create(conn_config, home_dir);
+    WT_CONNECTION *conn = connection_manager::instance().get_connection();
+
+    /* Open different sessions. */
+    WT_SESSION *insert_session, *read_session;
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &insert_session));
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &read_session));
+
+    /* Create a collection. */
+    const std::string collection_name = "table:my_collection";
+    testutil_check(
+      insert_session->create(insert_session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
+
+    /* Open different cursors. */
+    WT_CURSOR *insert_cursor, *read_cursor;
+    const std::string cursor_config = "";
+    testutil_check(insert_session->open_cursor(
+      insert_session, collection_name.c_str(), nullptr, cursor_config.c_str(), &insert_cursor));
+    testutil_check(read_session->open_cursor(
+      read_session, collection_name.c_str(), nullptr, cursor_config.c_str(), &read_cursor));
+
+    /* Store cursors. */
+    std::vector<WT_CURSOR *> cursors;
+    cursors.push_back(insert_cursor);
+    cursors.push_back(read_cursor);
+
+    /* Insert some data. */
+    std::string key = "a";
+    const std::string value = "b";
+    insert_cursor->set_key(insert_cursor, key.c_str());
+    insert_cursor->set_value(insert_cursor, value.c_str());
+    testutil_check(insert_cursor->insert(insert_cursor));
+
+    /* Read some data. */
+    key = "b";
+    read_cursor->set_key(read_cursor, key.c_str());
+    testutil_assert(read_cursor->search(read_cursor) == WT_NOTFOUND);
+
+    key = "a";
+    read_cursor->set_key(read_cursor, key.c_str());
+    testutil_check(read_cursor->search(read_cursor));
+
+    /* Create a thread manager and spawn some threads that will work. */
+    thread_manager t;
+    int key_size = 1, value_size = 2;
+
+    do_inserts = true;
+    t.add_thread(insert_op, insert_cursor, key_size, value_size);
+
+    do_reads = true;
+    t.add_thread(read_op, read_cursor, key_size);
+
+    /* Sleep for the test duration. */
+    int test_duration_s = 5;
+    std::this_thread::sleep_for(std::chrono::seconds(test_duration_s));
+
+    /* Stop the threads. */
+    do_reads = false;
+    do_inserts = false;
+    t.join();
+
+    /* Close cursors. */
+    for (auto c : cursors)
+        testutil_check(c->close(c));
+
+    /* Close the connection. */
+    connection_manager::instance().close();
```

Comment:
I'm not saying it needs to be implemented now, but usually, with this type of classes that need to do some cleanup at the end of the main, it's useful to have a wrapper class that is created on the stack and in the destructor it does the cleanup. It would call `connection_manager::instance().create()` in the constructor and call `connection_manager::instance().close()` in the destructor. It could be called `ConnectionWrapper` or something like that.

Reply:
I think this is what we tried with our usual cpp tests [here](https://github.com/wiredtiger/wiredtiger/blob/develop/test/cppsuite/test_harness/test.h#L101-L106). But as you can see it is a workaround. We diddn't do more regarding this issue as we were thinking of reworking the `connection_manager` class to be able to maintain multiple connections.
But that's effectively a good idea!